tests/test_sign_with_hsm/functions.sh

```diff
@@ -79,7 +79,22 @@ EOF
         -v menderartifact="${mender_artifact}" \
         -v artifact="$artifact" \
         '/URL/{ rc=system(menderartifact" sign --key-pkcs11 \""$NF";pin-value="pin"\" "artifact); exit(rc); }' && log PASSED
-    "${mender_artifact}" \
-    read \
-    "$artifact" | awk '/Signature: /{ if ($2=="signed") { exit(0); } else { exit(1); } }'
+    p11tool \
+        --login \
+        --provider=/usr/lib/softhsm/libsofthsm2.so \
+        --set-pin="$pin" \
+        --list-all-privkeys | awk \
```

Comment:
For `validate` and `read` commands what we need is the public key. I assume this command to `p11tool` returns the private keys, so how is it supposed to work?

Reply:
This is how it works with both SoftHSM and other PKCS#11 implementations. It is counter-intuitive, but as a private key always embeds the public key, you have the information to perform the validation. Additionally, what you store in the softHSM is the private key, not the public.